### PR TITLE
Fixing Navbar

### PR DESCRIPTION
**Pull request by EvanZhouDev**
Improved the navbar (at least made it work)

Features
- Added dynamic navbar
- Navbar buttons work
- Added visual indicators to each HTML page
- Added animations to test the script running

Potential future features:
- Github link on navbar
- Better menu?

### CODE REVIEW

**Comment by EvanZhouDev** [comment]
I realize that Bryan also has done an implementation. However...
- Similar implementation, his is done with ternaries
- His navbar buttons don't work. (I know he will probably fix this, but I already implemented it.)
- His navbar doesn't loop back. ("Next" part is blank when you get to the end on his, while mine shows the first chapters)

Please take this into consideration while approving my PR.

File: src/acts-api.js [hunk 1]
```diff
@@ -2,40 +2,64 @@
 const SID_MAP = Object.fromEntries($store.pages.map((x, i) => [x.sID, i]));
 
 export function switchPage(sID) {
-  if (!sID in SID_MAP) {
-    console.error("Please input valid sID to switch to. Action aborted.");
-    return;
-  }
-  let currentPageNum = JSON.parse(JSON.stringify(SID_MAP[sID]));
+	if (!sID in SID_MAP) {
```
Comment (ThatXliner) [inline src/acts-api.js:5]: Ok, what’s with the formatting? I thought we were just using config-less prettier?

File: src/acts-api.js [hunk 2]
```diff
@@ -2,40 +2,64 @@
...
+		.then((data) => {
+			// Safer than eval; also faster.
+			Function(data)();
+		});
 
+	// Update the navbar
+	let navbarPrev = document.getElementById("navbarPrev");
+	let navbarNow = document.getElementById("navbarNow");
+	let navbarNext = document.getElementById("navbarNext");
+	let pages = window.$store.pages;
```
Comment (ThatXliner) [inline src/acts-api.js:36]: You don’t need to specify `window.$store`; `$store` Works

File: src/acts-api.js [hunk 3]
```diff
@@ -2,40 +2,64 @@
...
+	if (currentPageNum == pages.length - 1) {
+		navbarNext.innerHTML = parseSID(pages[0].sID);
```
Comment (ThatXliner) [inline src/acts-api.js:38]: These should be innerText or textContent

File: src/acts-api.js [hunk 4]
```diff
@@ -2,40 +2,64 @@
...
+	} else {
+		navbarPrev.innerHTML = parseSID(pages[currentPageNum - 1].sID);
+		navbarPrev.setAttribute("goto", pages[currentPageNum - 1].sID);
+		navbarNext.innerHTML = parseSID(pages[currentPageNum + 1].sID);
+		navbarNext.setAttribute("goto", pages[currentPageNum + 1].sID);
+	}
+	navbarNow.innerHTML = parseSID(pages[currentPageNum].sID);
+	navbarNow.setAttribute("goto", pages[currentPageNum].sID);
+}
+switchPage("acts1_3");
```
Comment (ThatXliner) [inline src/acts-api.js:56]: Is this for initial load? Do that in `setup.js`. Also add a comment too

File: src/acts-api.js [hunk 5]
```diff
@@ -2,40 +2,64 @@
...
+		navbarNext.setAttribute("goto", pages[currentPageNum + 1].sID);
+	} else {
+		navbarPrev.innerHTML = parseSID(pages[currentPageNum - 1].sID);
+		navbarPrev.setAttribute("goto", pages[currentPageNum - 1].sID);
+		navbarNext.innerHTML = parseSID(pages[currentPageNum + 1].sID);
+		navbarNext.setAttribute("goto", pages[currentPageNum + 1].sID);
+	}
+	navbarNow.innerHTML = parseSID(pages[currentPageNum].sID);
+	navbarNow.setAttribute("goto", pages[currentPageNum].sID);
+}
```
Comment (ThatXliner) [inline src/acts-api.js:55]: Implementation looks good. My implementation uses web components in case we wanted to reference to a page within the animation

**Review by ThatXliner — COMMENTED** [review]
I guess changing the goto attribute works since our onclick event handler dynamically gets the goto attribute. I wanted to try to switch using web components instead of `a` links + `goto` because
1. See my review comment
2. Possibly more maintainable

overall, seems pretty good and functional (I haven’t tested locally but there doesn’t seem to be any bugs) besides some nitpicking.

File: src/acts-api.js (hunk 3 above)
Comment (ThatXliner) [reply src/acts-api.js:38]: ~~textContent is slightly more performant~~

**Comment by EvanZhouDev** [comment]
Refer to @ThatXliner's navbar commit.

**Pull request closed without merging**